src/validations/rules/Writing_FedRAMP_Validations.md

```diff
@@ -0,0 +1,375 @@
+
+Constraints define required or recommended data elements FedRAMP expects in syntactically valid OSCAL documents. When developing rules, developers will recognize in most cases there are two pieces to a rule checking the constraint: the analytical query of the OSCAL data, and cross-referencing that data with a set of known values for given data elements in a constraint. Much like other software projects, FedRAMP has opted to store the known values as external configuration that can be adapted with changing the analytical query operation of the rule. These are the FedRAMP values.
+
+Schematron constraints have thus far been for system security plan documents. They are found in a Schematron document [`ssp.sch`](ssp.sch).
+
+A companion document [`fedramp_values.xml`](../../dist/content/resources/xml/fedramp_values.xml) defines sets of allowed values for various FedRAMP OSCAL XML elements and their attributes.
+
+An additional Schematron document [`sch.sch`](../sch/sch.sch) defines a style guide for FedRAMP Schematron to ensure developers continue to follow best practices. These augmentations provide
+
+- References to associated unit tests ([ssp.xspec](../test/ssp.xspec))
+- References to FedRAMP documentation related to the constraint
+- References to associated diagnostic messages
+- Forward and reverse linkages between Schematron rules (`<rule>` elements), assertions (`<assert>` elements, and diagnostic messages (`<diagnostic>` elements).
+
+Use a validating (XML) editor with the `sch.sch` employed in order to enforce the style.
+
+# Examples of Schematron constraints
+
+The following examples use XML namespace prefixes declared at the root element of the Schematron document.
+
+```
+<sch:schema
+    queryBinding="xslt2"
+    xmlns:array="http://www.w3.org/2005/xpath-functions/array"
+    xmlns:doc="https://fedramp.gov/oscal/fedramp-automation-documentation"
+    xmlns:map="http://www.w3.org/2005/xpath-functions/map"
+    xmlns:o="http://csrc.nist.gov/ns/oscal/1.0"
+    xmlns:sch="http://purl.oclc.org/dsdl/schematron"
+    xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
+
+    <sch:ns
+        prefix="f"
+        uri="https://fedramp.gov/ns/oscal" />
+    <sch:ns
+        prefix="o"
+        uri="http://csrc.nist.gov/ns/oscal/1.0" />
+    <sch:ns
+        prefix="oscal"
+        uri="http://csrc.nist.gov/ns/oscal/1.0" />
+    <sch:ns
+        prefix="fedramp"
+        uri="https://fedramp.gov/ns/oscal" />
+    <sch:ns
+        prefix="lv"
+        uri="local-validations" />
+    <sch:ns
+        prefix="array"
+        uri="http://www.w3.org/2005/xpath-functions/array" />
+    <sch:ns
+        prefix="map"
+        uri="http://www.w3.org/2005/xpath-functions/map" />
+``` 
+Namespace declarations made on the root element are for that document.
+
+Namespace declarations via the `<sch:ns>` element are for instance documents to be validated.
+
+## Example 1 - Existential and value checks
```

Comment:
"Existential" seems cumbersome for this concept.   How about "Logic and value checks"  .. Just MHO.

Reply:
Altered a bit to explain the specific test for existence.